File: Activities Java/ArraysActivities/MethodsOfOrden/QuickSort.java
public class QuickSort {

    static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    static int partition(int[] arr, int l, int h) {
        int pvt = arr[h]; 
        int j = l - 1;
        for (int k = l; k < h; k++) {
            if (arr[k] < pvt) {
                j++;
                swap(arr, j, k);
            }
        }
        swap(arr, j + 1, h); 
        return j + 1;
    }

    // QuickSort recursivo
    static void quickSort(int[] arr, int l, int h) {
        if (l < h) {
            int pi = partition(arr, l, h);
            quickSort(arr, l, pi - 1);
            quickSort(arr, pi + 1, h);
        }
    }

    // Método principal
    public static void main(String[] args) {
        int[] arr = {10, 7, 8, 9, 1, 5};
        int size = arr.length;

        System.out.println("El arreglo antes de ordenarlo:");
        for (int v : arr) {
            System.out.print(v + " ");
        }
        System.out.println();

        quickSort(arr, 0, size - 1);

        System.out.println("El arreglo después de ordenarlo:");
        for (int v : arr) {
            System.out.print(v + " ");
        }
        System.out.println();
    }
}

File: Activities C++/output/MethodsOfOrden/QuickSort.cpp
#include <iostream>
using namespace std;

void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}

int partition(int arr[], int l, int h) {
    int pvt = arr[h];  // Último elemento como pivote
    int j = l - 1;
    for (int k = l; k < h; k++) {
        if (arr[k] < pvt) {
            j++;
            swap(arr[j], arr[k]);
        }
    }
    swap(arr[j + 1], arr[h]);  // Coloca el pivote en su lugar correcto
    return j + 1;
}

void qckSort(int arr[], int l, int h) {
    if (l < h) {
        int pi = partition(arr, l, h);
        qckSort(arr, l, pi - 1);
        qckSort(arr, pi + 1, h);
    }
}

int main() {
    int arr[] = {10, 7, 8, 9, 1, 5};
    int size = sizeof(arr) / sizeof(arr[0]);

    cout << "El arreglo antes de ordenarlo: ";
    for (int i = 0; i < size; i++)
        cout << arr[i] << " ";
    cout << endl;

    qckSort(arr, 0, size - 1);

    cout << "El arreglo después de ordenarlo: ";
    for (int i = 0; i < size; i++)
        cout << arr[i] << " ";
    cout << endl;

    return 0;
}
